src/Workspaces/CoreTestUtilities/MEF/ExportProviderCache.cs

```diff
@@ -200,6 +203,33 @@ public ExportProvider GetOrCreateExportProvider()
                 var expected = _expectedProviderForCatalog;
                 if (expected == null)
                 {
+                    foreach (var errorCollection in _configuration.CompositionErrors)
+                    {
+                        foreach (var error in errorCollection)
+                        {
+                            foreach (var part in error.Parts)
+                            {
+                                foreach (var (importBinding, exportBindings) in part.SatisfyingExports)
+                                {
+                                    if (exportBindings.Count <= 1)
+                                    {
+                                        // Ignore composition errors for missing parts
+                                        continue;
+                                    }
+
+                                    if (importBinding.ImportDefinition.Cardinality != ImportCardinality.ZeroOrMore)
+                                    {
+                                        // This failure occurs when a binding fails because multiple exports were
+                                        // provided but only a single one (at most) is expected. This typically occurs
+                                        // when a test ExportProvider is created with a mock implementation without
+                                        // first removing a value provided by default.
+                                        Assert.Null(error.Message);
```

Comment:
Why not just `throw new Exception(error.Message)`? Two small reasons:

1. The "expected null, actual: message" is confusing. We don't expect a null message. We expect to not be here.
2. If MEF ever gave a null message (even if that's a bug), then your assert won't fire and now you have two bugs.

(This mostly threw up a flag in my head because I commonly see people not wanting to just throw an exception in xUnit and workaround it with Assert.False(true, "message") or other things that get kinda confusing too.)